### PR TITLE
Add mmp support for AOT options in XS

### PR DESCRIPTION
**Pull request by timrisi**
This is needed to be able to add the AOT options in the xamarin.mac build options in xamarin studio

### CODE REVIEW

File: msbuild/Xamarin.Mac.Tasks.Core/Tasks/MmpTaskBase.cs [hunk 1]
```diff
@@ -72,6 +72,10 @@ public class MmpTaskBase : ToolTask
 		public string I18n { get; set; }
 		public string ExtraArguments { get; set; }
 
+		public string AOTMode { get; set; }
+		public bool HybridAOT { get; set; }
```
Comment (spouliot) [inline msbuild/Xamarin.Mac.Tasks.Core/Tasks/MmpTaskBase.cs]: it sounds a bit confusing to to have `AOTMode` and then `HybridAOT` which is a _mode_

Maybe seeing a pic of the UI would help, but I think the terms could be clearer

**Comment by timrisi** [comment]
screen shot 2017-03-02 at 12 53 39 pm

AOTMode is the dropdown, for None, Core, SDK or All.  If it's set to anything but None, the checkbox and textbox are enabled.  HybridAOT for the checkbox defaults to false, textbox defaults to empty (I'd been messing with the options in the above SS).

I'm definitely open to changing the variable names.

**Comment by spouliot** [comment]
Ok, I think `Mode` should be reserved for `Normal`, `Hybrid` or `Full` (even if the last does not apply here today). It's already the term used between `normal` and `full`  AOT builds on iOS.

It seems to me that `Scope` or `Set` would better terms, as the option defines a predefined set of assemblies where AOT would be applied.

I'm not sure a text box is a good UI for listing assemblies, even more if it deals with both addition and removals (it means you have to know the syntax). There's already a text box for **Additional mmp arguments**  that can be used for *advanced options*. Maybe that text box can be removed ?

@VincentDondain is there another way (e.g. in referenced assemblies) that we could show/select assemblies for "AOT > No, Normal, Hybrid" ?

**Comment by spouliot** [comment]
hmmm, the last part (of my previous comment) was not written is the right PR. Let's just focus on my two first paragraphs for this PR :-)

File: msbuild/Xamarin.Mac.Tasks.Core/Tasks/MmpTaskBase.cs (hunk 1 above)
Comment (VincentDondain) [inline msbuild/Xamarin.Mac.Tasks.Core/Tasks/MmpTaskBase.cs]: I'd go for `AOTMode` and `HybridAOTOption`

Comment (timrisi) [inline msbuild/Xamarin.Mac.Tasks.Core/Tasks/MmpTaskBase.cs]: AOTScope maybe (based on sebastians comment below)?

Related: Should it just be AotScope, etc for the names instead?  Similar to how we rename NSURL names to NSUrl

**Comment by VincentDondain** [comment]
I agree with @spouliot on the fact that "Include/exclude assemblies from AOT compilation" is a bit too advanced and could be reserve for use in the additional mmp arguments (simplifying the build options).

@spouliot about the last comment, you're talking about that?

screenshot 2017-03-02 20 18 35

Like we'd right click on some assemblies and select an AOT mode? Kinda like build actions on files?

**Comment by timrisi** [comment]
The UI discussion should probably move to the PR where those are actually implemented

**Comment by VincentDondain** [comment]
Ah damn I didn't notice we were in `xamarin-macios` I just jumped on the comments from my emails (:

**Comment by VincentDondain** [comment]
Moved the UI conversation to https://github.com/xamarin/md-addins/pull/1511#issuecomment-283837282

**Comment by timrisi** [comment]
Failure unrelated: https://bugzilla.xamarin.com/show_bug.cgi?id=52538

**Comment by spouliot** [comment]
@timrisi you can merge (like it was a green checkmark) once you identified (or filed) the unrelated failure

**Pull request merged**